### PR TITLE
Docs: Jest usage

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Heavily inspired by the solution of @AlanFoster and with the help of @levito

### CODE REVIEW

File: docs/Guide.Jest.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  await device.reloadReactNative();
+});
+```
+
+### 4. Run jest
+
+Add this part to your `package.json`:
+```json
+"scripts": {
+    "test:e2e": "jest e2e --setupTestFrameworkScriptFile=./jest/setup-e2e-tests.js"
```
Comment (chirag04) [inline docs/Guide.Jest.md]: I like how you can define jest config in your package.json so you don't have to pass in `--setupTestFrameworkScriptFile`. Just making a note. either way is fine. 

```
"jest": {
    "bail": true,
    "setupTestFrameworkScriptFile": "<rootDir>/jest/setup.js"
  },
```

**Comment by rotemmiz** [comment]
I would add a note re Detox that it exports expect, waitFor, element and by, so Jest users might find themselves using a different expect object implicitly.

**Comment by rotemmiz** [comment]
BTW, we plan on adding another detox.init flavour/flag that will not export all that into globals, and users would need to explicitly import it. It would make lots of sense for Jest.

That might be an AWESOME opportunity for a PR (anyone? hint hint...)

File: docs/Guide.Jest.md (hunk 1 above)
Comment (DanielMSchmidt) [reply docs/Guide.Jest.md]: This is intentional because we wanted to make sure that they run side by side with your normal unit tests

Comment (chirag04) [reply docs/Guide.Jest.md]: sure. either way is fine. you can push that logic in setup file too: https://github.com/grabbou/react-native-detox-example/blob/master/jest/setup.js#L10-L14

**Comment by DanielMSchmidt** [comment]
@rotemmiz Noted the exposure of expect and so on
@chirag04 Thanks for the note, but this was intentional to have test run side by side with the unit tests

File: docs/Guide.Jest.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+We need the `--runInBand` as detox doesn't support parallelism yet.
+
+### Writing Tests
+
+There are some things you should notice:
+
+- Don't worry about mocks being used, detox works on the compiled version of your app.
+- Detox exposes it's primitives (`expect`, `device`, ...) globally
```
Comment (rotemmiz) [inline docs/Guide.Jest.md]: Detox exposes it's primitives (`expect`, `device`, ...) globally, it will override Jest's global `expect` object.

**Comment by DanielMSchmidt** [comment]
@rotemmiz I will merge this, implemented your wish. Feel free to edit and improve 👍

**Pull request merged**